packages/open-next/src/http/openNextResponse.ts

```diff
@@ -384,4 +386,20 @@ export class OpenNextNodeResponse extends Transform implements ServerResponse {
     //TODO: test to see if we need to call end here
     return this;
   }
+
+  // For some reason, next returns the 500 error page with some cache-control headers
+  // We need to fix that
+  private fixHeadersForError() {
+    if (process.env.OPEN_NEXT_DANGEROUSLY_SET_ERROR_HEADERS === "true") {
+      return;
+    }
+    // We only check for 404 and 500 errors
+    // The rest should be errors that are handled by the user and they should set the cache headers themselves
+    if (this.statusCode === 404 || this.statusCode === 500) {
```

Comment:
Do we want to make this fix on ALL status except `2xx` and `301` (permanent redirect)?

Reply:
I don't think so, this is a very special case of Next overwriting the cache-control and it could only happen for 404 and 500.
There is the case of 307/308 by redirecting in ISR but those are ISR so we expect the status code.

The rest of the status code are user adding those themselves. And they should be able to do so if they wish to but that's their responsability